src/main/scala/cognite/spark/v1/SdkV1Rdd.scala

```diff
@@ -45,82 +45,22 @@ case class SdkV1Rdd[A, I](
 
     val streams = getStreams(client, config.limitPerPartition, config.partitions)
     val groupedStreams = streams.grouped(config.parallelismPerPartition).toSeq
-    val currentStreamsAsSingleStream = groupedStreams(split.index)
-      .reduce(_.merge(_))
-
-    // This pool will be used for draining the queue
-    // Draining needs to have a separate pool to continuously drain the queue
-    // while another thread pool fills the queue with data from CDF
-    val drainPool = Executors.newFixedThreadPool(1)
-    val drainContext = ExecutionContext.fromExecutor(drainPool)
-
-    // Local testing show this queue never holds more than 5 chunks since CDF is the bottleneck.
-    // Still setting this to 2x the number of streams being read to makes sure this doesn't block
-    // too early, for example in the event that all streams return a chunk at the same time.
-    val queue = new EitherQueue(config.parallelismPerPartition * 2)
-
-    val putOnQueueStream =
-      enqueueStreamResults(currentStreamsAsSingleStream, queue)
-        .handleErrorWith(e => Stream.eval(IO(queue.put(Left(e)))) ++ Stream.raiseError[IO](e))
-
-    // Continuously read the stream data into the queue on a separate thread pool
-    val streamsToQueue: Future[Unit] = Future {
-      putOnQueueStream.compile.drain.unsafeRunSync()
-    }(drainContext)
-
-    queueIterator(queue, streamsToQueue) {
-      if (!drainPool.isShutdown) {
-        drainPool.shutdown()
-      }
+    val currentStreamsAsSingleStream = config.limitPerPartition match {
+      case Some(limit) => groupedStreams(split.index).map(_.take(limit)).reduce(_.merge(_))
```

Comment:
Is this `take` really necessary, as it's applied to each "CDF partition", however we have also set the limit at line 37.

Reply:
Nope! Good catch, that already take cares of limitPerPartition for SdkV1Rdd.